### PR TITLE
Update Operator SDK to 1.21.0

### PR DESCRIPTION
**Pull request by johnxo-rh**
Update operator-sdk to v1.21.0
https://issues.redhat.com/browse/OSD-12091

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -1,23 +1,27 @@
 FROM quay.io/app-sre/boilerplate:image-v2.3.2 AS builder
+FROM golang:1.17 as builder
```
Comment (wanghaoran1988) [inline build/Dockerfile]: The boilerplate already have golang, you should remove this.

File: README.md [hunk 2]
```diff
@@ -1,238 +1,94 @@
-# AWS EFS Operator for OpenShift Dedicated
```
Comment (2uasimojo) [inline README.md]: It looks like this README was replaced by a stock/generated one. Is there a reason the old content is no good?

**Review by 2uasimojo — COMMENTED** [review]
I didn't go through this in depth, but I see a *lot* of changes I don't understand.

A lot of test cases/paths have been removed or commented out -- why is that?

Some new files were introduced that seem like they should be part of an import, like mapper.go, k8sutil.go, logger.go

Some files were deleted or replaced by stock versions, like the README and sample SharedVolume manifests (should the latter just have been moved instead?).

---

On a more general note, I don't know if we're even supporting this operator at all anymore. @jharrington22 @dustman9000 ?

File: README.md (hunk 2 above)
Comment (jharrington22) [inline README.md]: Why is this README changing? I don't see the contents anywhere else?

File: pkg/util/ensurable_test.go [hunk 3]
```diff
@@ -11,7 +11,9 @@ import (
 
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
```
Comment (jharrington22) [inline pkg/util/ensurable_test.go:14]: The naming of this is confusing, `crclient` to me means CodeReady Containers (https://developers.redhat.com/products/openshift-local/overview)

File: pkg/util/predicate_test.go [hunk 4]
```diff
@@ -112,105 +111,28 @@ func TestUpdate(t *testing.T) {
...
+				/*removed in newer version*/
+				// MetaOld:   mkTestMeta(oldMeta),
+				// MetaNew:   mkTestMeta(newMeta),
 			},
 			want,
 		}
 	}
 	tests := []test{
 		// We care when *either* the old or new passes
-		mktest(care, care, care, care, true),
```
Comment (jharrington22) [inline pkg/util/predicate_test.go:131]: Why are these test cases being removed?

File: pkg/util/predicate_test.go [hunk 5]
```diff
@@ -33,13 +33,13 @@ func mkTestObj(kind testObjKind) runtime.Object {
...
 // mkTestObj returns a new metav1.Object that's nil, or that our predicate methods should either
 // care about or not, per the value of `kind`.
-func mkTestMeta(kind testObjKind) metav1.Object {
-	obj := mkTestObj(kind)
-	if obj == nil {
-		return nil
-	}
-	return obj.(metav1.Object)
-}
+// func mkTestMeta(kind testObjKind) metav1.Object {
```
Comment (jharrington22) [inline pkg/util/predicate_test.go:36]: Is this metadata not needed anymore?

File: controllers/statics/statics_test.go [hunk 6]
```diff
@@ -109,33 +110,7 @@ func TestEnsureStatics(t *testing.T) {
 
 // TestEnsureStaticsError tests the error path of EnsureStatics
 func TestEnsureStaticsError(t *testing.T) {
```
Comment (jharrington22) [inline controllers/statics/statics_test.go:112]: Why are these tests removed?

**Review by jharrington22 — CHANGES_REQUESTED** [review]
I left some comments but similar concerns to @2uasimojo

**Comment by jharrington22** [comment]
> I didn't go through this in depth, but I see a _lot_ of changes I don't understand.
> 
> A lot of test cases/paths have been removed or commented out -- why is that?
> 
> Some new files were introduced that seem like they should be part of an import, like mapper.go, k8sutil.go, logger.go
> 
> Some files were deleted or replaced by stock versions, like the README and sample SharedVolume manifests (should the latter just have been moved instead?).
> 
> On a more general note, I don't know if we're even supporting this operator at all anymore. @jharrington22 @dustman9000 ?

We don't have a migration path from this to the new OCP supported operator so we're updating the SDK to keep it inline with our development environment so that if we need to do any maintenance it shouldn't be difficult. This will also allow us to update controller runtime deps etc.

**Comment by dustman9000** [comment]
/hold 
Currently aws-efs-operator is no longer supported by RH. There has been work done with [SDE-1759](https://issues.redhat.com/browse/SDE-1759) to close this gap and provide a migration path for customers to use the supported operator.

File: README.md (hunk 2 above)
Comment (johnxo-rh) [reply README.md]: I will update previous ReadMe and push

File: pkg/util/ensurable_test.go (hunk 3 above)
Comment (johnxo-rh) [reply pkg/util/ensurable_test.go:14]: This is the same naming standard followed in other operator upgrades. Eg: managed upgrade operator

File: pkg/util/predicate_test.go (hunk 5 above)
Comment (johnxo-rh) [reply pkg/util/predicate_test.go:36]: Yes

File: controllers/statics/statics_test.go (hunk 6 above)
Comment (johnxo-rh) [reply controllers/statics/statics_test.go:112]: Logger is not interface any more. Log.Expect () would not work.

File: pkg/util/predicate_test.go (hunk 4 above)
Comment (johnxo-rh) [reply pkg/util/predicate_test.go:131]: These are redundant test cases

**Pull request closed without merging**